### ISSUE TITLE
MissingUseCall false positive with precondition functions

### ISSUE THREAD

**Issue by matejdro** [detekt/detekt#9122]
## Expected Behavior

Following should pass, because we are using `use`:

```kotlin
fun createInputStream(): InputStream? {
   throw UnsupportedOperationException("Only for detekt")
}

requireNotNull(createInputStream()) { "Stream should not be null at this point" }.use {
}
```

## Observed Behavior

`MissingUseCall` rule is triggered

## Context

I understand detekt flagging this, as `requireNotNull` function could theoretically use the stream before returning it, but in this case this does not happen.

Would having a configurable list of functions work for fixing this? Or maybe just hardcoded list of built-in stdlib precondition functions? Or is this so rare that it does not warrant fixing?

## Your Environment

* Version of detekt used: 2.0.0-alpha02
